var/spack/repos/builtin/packages/vtk-m/package.py

```diff
@@ -227,13 +227,11 @@ def cmake_args(self):
         return options
 
     # Delegate in the vtk-m built smoke test
-    def smoke_test(self):
+    def test_smoke_test(self):
         spec = self.spec
 
         if "+examples" not in spec:
-            raise RuntimeError(
-                "Examples needed for smoke test missing", "reinstall with `+examples` variant"
-            )
+            raise SkipTest("Package must be installed with 'examples'")
```

Comment:
Be explicit that `examples` is a boolean variant:

```suggestion
            raise SkipTest("Package must be installed with '+examples'")
```